[Guide - The New Architecture] Pillars (#3046)
facebook/react-native-website: docs/the-new-architecture/pillars.md

title: What Compose the New Architecture
---

> The new architecture is composed mainly by two pillars:

Reviewer: As a rule of thumb, let's use New Architecture (title case) everywhere as it's like the name of a product

- [TurboModules](pillars-turbomodules)
- [Fabric Components](pillars-fabric-components).

> TurboModules are the preferred way to create libraries that leverage some platform specific API. Fabric Components are the preferred way to create reusable UI components t provide a native experience to the users.

Reviewer: ```suggestion
TurboModules are the preferred way to create libraries that leverage some platform specific API. Fabric Components are the preferred way to create reusable UI components that provide a native experience to the users.
```

The main goal of this section is to drive the reader through a step-by-step guide to create their first TurboModule or Fabric Component.

The next sections contain an high-level overview of the pillars, together with the steps to create them. To create one of these pillars, the steps are:

1. Define a JavaScript specification using Flow or TypeScript.

> 1. Configure the dependencies management system to properly install the pillar.

Reviewer: This reads odd. I would not refer to "the pillar" as I consider it as an abstract concept.

```suggestion
1. Configure the dependencies management system to generate code from the provided spec
```

1. Implement the Native code.

> 1. Integrate the pillar in the app.

Reviewer: ```suggestion
1. Integrate the code in the app.
```

or whatever other term

Finally, we dive a little deeper into the [CodeGen](pillars-codegen) process that is required to create all the C++ types and files used by our components, including some useful steps to work comfortably while developing the component.

:::caution

> To integrate the pillars in an app, the app has to run with the New Architecture enabled.

Reviewer: ```suggestion
To integrate a TurboModule or a Fabric Component in an app, the app has to run with the New Architecture enabled.
```

To create a new app adopting the New Architecture, refer to the [Using the App Template](use-app-template) section.

> To migrate an existing app to the New Architecture, refer to the [Migration](../new-architecture-intro) guide.

Reviewer 2: ```suggestion
To migrate an existing app to the New Architecture, refer to the [Migration](/docs/new-architecture-library-intro) guide.
```

Let's use full path instead of relative one in links if possible. This make easier for Docusuaurs to detect broken links, when the path changes.

> To migrate an existing app to the New Architecture, refer to the [Migration](/docs/new-architecture-library-intro) guide.

Reviewer 2: ```suggestion
To migrate an existing app to the New Architecture, refer to the [Migration](/docs/new-architecture-intro) guide.
```

Whoops, it looks like you have accepted the change, before I have edited the link. 😅 I the first version, by mistake I have added `-library` to the path (different page).

Author: Fixed!
